README.md

```diff
@@ -53,9 +53,31 @@ Combining `editor.suggest.preview` with the [pre-fill required fields](#code-com
 
 Completing the snippet allows you to tab complete through each attribute and block.
 
-### Syntax Validation
+### Validation
+
+While editing, the configuration is validated and invalid code is highlighted. 
+
+HCL syntax is checked for e.g. missing `}`, `"` or other "control characters" being out of place.
+
+![](docs/validation-rule-hcl.png)
+
+Enhanced validation of selected Terraform language constructs in both `*.tf` and `*.tfvars` files based on detected Terraform version and provider versions is also available from `v2.28.0`. This can highlight deprecations, missing required attributes or blocks, references to undeclared variables and more, [as documented](https://github.com/hashicorp/terraform-ls/blob/main/docs/validation.md#enhanced-validation).
+
+![](docs/validation-rule-missing-attribute.png)
+
+![](docs/validation-rule-invalid-ref.png)
+
+The enhanced validation is on by default but can be turned off using the following settings option:
+
+```json
+"terraform.validation.enableEnhancedValidation": false
+```
+
+The extension also provides validation through [`terraform validate`](https://www.terraform.io/cli/commands/validate). This can be triggered via command palette. Unlike the other validation methods, this one implies Terraform CLI being installed and prior successful run of `terraform init` (i.e. local installation of all providers and modules). It is the slowest method, but the most thorough - i.e. it will catch the most mistakes.
```

Comment:
```suggestion
The extension also provides validation through [`terraform validate`](https://www.terraform.io/cli/commands/validate). This can be triggered via command palette. Unlike the other validation methods, this one requires the Terraform CLI installed and a previous successful run of `terraform init` (i.e. local installation of all providers and modules) to function correctly. It is the slowest method, but the most thorough - i.e. it will catch the most mistakes.
```